### PR TITLE
feat: form parse support custom qs module

### PR DESCRIPTION
**Pull request by fengmk2**

### CODE REVIEW

**Comment by fengmk2** [comment]
cc @fool2fish

File: test/form.js [hunk 1]
```diff
@@ -58,4 +58,29 @@ describe('parse.form(req, opts)', function(){
...
+          err.status.should.equal(400);
+          err.message.should.equal('Index of array [21] is overstep limit: 20');
+          this.body = err.message;
+        }
+      });
+
+      request(app.listen())
+      .post('/')
+      .type('form')
+      .send({ a: { '21': 'a' } })
```
Comment (coderhaoxin) [inline test/form.js]:
```js
.expect('Index of array [21] is overstep limit: 20')
```

Comment (fengmk2) [inline test/form.js]: fixed

**Comment by fengmk2** [comment]
4.1.0

**Pull request merged**